integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/vector/LuceneVectorFieldIT.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.lucene.vector;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.search.integrationtest.backend.lucene.testsupport.util.DocumentAssert.containsDocument;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.hibernate.search.backend.lucene.LuceneExtension;
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.extension.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import org.apache.lucene.document.Document;
+
+class LuceneVectorFieldIT {
+
+	private static final byte[] BYTE_VECTOR_1 = new byte[] { 1, 2, 3, 4 };
+	private static final byte[] BYTE_VECTOR_2 = new byte[] { 1, 1, 1, 1 };
+
+	private static final float[] FLOAT_VECTOR_1 = new float[] { 1.0f, 2.0f, 3.0f, 4.0f, 1.0f, 2.0f, 3.0f, 4.0f };
+	private static final float[] FLOAT_VECTOR_2 = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f };
+
+	@RegisterExtension
+	public final SearchSetupHelper setupHelper = SearchSetupHelper.create();
+
+	private final SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new );
+
+	@BeforeEach
+	void setup() {
+		setupHelper.start()
+				.withIndex( index )
+				.setup();
+		initData();
+	}
+
+	@Test
+	void simpleVectorSavedAndRetrieved() {
+		SearchQuery<Document> query = index.createScope().query()
+				.select(
+						f -> f.extension( LuceneExtension.get() ).document()
+				)
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<Document> result = query.fetchAll().hits();
+		assertThat( result )
+				.hasSize( 2 )
+				.satisfies( containsDocument(
+						doc -> doc.hasField( "string", "keyword1" )
+								.hasField( "byteVector", Arrays.toString( BYTE_VECTOR_1 ) )
+								.hasField( "floatVector", Arrays.toString( FLOAT_VECTOR_1 ) )
```

Comment:
Or -- hear me out -- don't allow projections on vector fields :D 

Any of the above is fine to me.

Reply:
OK, I've tried to "serialize" floats as a byte array, and it doesn't look too terrible 😃, so maybe we can keep the projection. I think it might be helpful in some cases when the user wants to navigate the results going from a found item to a list of similar items.